documentation/src/main/doc/modules/kafka/examples/inbound/KafkaBatchRecordExample.java

```diff
@@ -0,0 +1,46 @@
+package inbound;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+
+import io.smallrye.reactive.messaging.kafka.KafkaBatchRecords;
+import io.smallrye.reactive.messaging.kafka.KafkaRecord;
+import io.smallrye.reactive.messaging.kafka.api.IncomingKafkaRecordMetadata;
+
+@ApplicationScoped
+public class KafkaBatchRecordExample {
+
+    @SuppressWarnings("unchecked")
+    // tag::code[]
+    @Incoming("prices")
+    public CompletionStage<Void> consumeMessage(KafkaBatchRecords<String, Double> records) {
+        for (KafkaRecord<String, Double> record : records) {
+            IncomingKafkaRecordMetadata<String, Double> metadata = record.getMetadata(IncomingKafkaRecordMetadata.class).orElse(null);
+            if (metadata != null) {
+                int partition = metadata.getPartition();
+                long offset = metadata.getOffset();
+                ConsumerRecord<String, Double> consumerRecord = metadata.getRecord();
+            }
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+
+    @Incoming("prices")
+    public CompletionStage<Void> consumeRecords(ConsumerRecords<String, Double> records) {
+        for (TopicPartition partition : records.partitions()) {
+            for (ConsumerRecord<String, Double> record : records.records(partition)) {
+
+            }
+        }
+        return CompletableFuture.completedFuture(null);
```

Comment:
Same here, how do we `ack`?

Reply:
I need to look more into this case, as `ConsumerRecords` is injected as part of a converter. I assume cases with current converters ack the message with post processing by default.

Reply:
I'd guess that this method should just return `void` then?

Or it could accept `Message<ConsumerRecords<String, Double>>` and `return msg.ack()` maybe. I didn't really think it through, maybe there's a problem with that.